nvflare/private/fed/server/server_engine.py

```diff
@@ -488,28 +493,49 @@ def send_aux_request(
             self.logger.error(f"Failed to send the aux_message: {topic} with exception: {secure_format_exception(e)}.")
 
     def sync_clients_from_main_process(self):
-        return_data = self._get_clients_data(self.args.job_id)
-        clients = return_data.get(ServerCommandKey.CLIENTS)
-        self.client_manager.clients = clients
-
-    def _get_clients_data(self, job_id):
+        # repeatedly ask the parent process to get participating clients until we receive the result
+        # or timed out after 30 secs (already tried 30 times).
         start = time.time()
+        max_wait = 30.0
```

Comment:
probably no need to use max_wait for 30 seconds. This should be relative fast.

Reply:
fast is fine since it won't wait that long - as soon as the result is retrieved, it will break out.

Reply:
that's fine.